tools/cldr-code/src/main/java/org/unicode/cldr/util/SupplementalDataInfo.java

```diff
@@ -5349,42 +5349,12 @@ public Set<String> getUnitPrefixes() {
                         @Override
                         public Set<String> get() {
                             Set<String> availableLongTz = sc.getAvailableCodes(CodeType.tzid);
-                            Set<String> result = null;
-                            if (true) { // hack for now
-                                final Set<String> hack =
-                                        Set.of(
-                                                "America/Santa_Isabel",
-                                                "Australia/Currie",
-                                                "America/Yellowknife",
-                                                "America/Rainy_River",
-                                                "America/Thunder_Bay",
-                                                "America/Nipigon",
-                                                "America/Pangnirtung",
-                                                "Europe/Uzhgorod",
-                                                "Europe/Zaporozhye",
-                                                "Pacific/Johnston");
-                                result = Set.copyOf(Sets.difference(availableLongTz, hack));
-                            } else { // TODO restore when CLDR-17412 is fixed
-                                Map<String, String> aliasToRegular =
-                                        bcp47KeyToAliasToSubtype.get("tz");
-                                Map<String, Bcp47KeyInfo> subtypeToInfo =
-                                        bcp47KeyToSubtypeToInfo.get("tz");
-                                result =
-                                        availableLongTz.stream()
-                                                .filter(
-                                                        x -> {
-                                                            String shortId = aliasToRegular.get(x);
-                                                            Bcp47KeyInfo info =
-                                                                    subtypeToInfo.get(shortId);
-                                                            System.out.println(
-                                                                    String.format(
-                                                                            "%s %s %s",
-                                                                            x, shortId, info));
-                                                            return !info.deprecated;
-                                                        })
-                                                .collect(Collectors.toUnmodifiableSet());
-                            }
-                            return result;
+                            return availableLongTz.stream()
+                                    .filter(
+                                            x -> {
+                                                return zone_territory.get(x) != null;
```

Comment:
I can understand dropping the hack. So that I can understand what is going on, can you explain why the old code had
```
return !info.deprecated;
```
and the new code just has 
```
return zone_territory.get(x) != null;
```

Reply:
The new code tries to check whether an ID is canonical (https://github.com/unicode-org/cldr/blob/10e099457512881f27d57785bcfadf4230b913af/tools/cldr-code/src/main/java/org/unicode/cldr/util/SupplementalDataInfo.java#L3470). A canonical ID is never deprecated.